src/modules/dwm.cpp

```diff
@@ -0,0 +1,493 @@
+#include "modules/dwm.hpp"
+
+#include <dwmipcpp/errors.hpp>
+
+#include "components/builder.hpp"
+#include "components/config.hpp"
+#include "drawtypes/label.hpp"
+#include "modules/meta/base.inl"
+#include "utils/env.hpp"
+#include "utils/file.hpp"
+#include "x11/extensions/randr.hpp"
+
+POLYBAR_NS
+
+namespace modules {
+  template class module<dwm_module>;
+
+  dwm_module::dwm_module(const bar_settings& bar, string name_) : event_module<dwm_module>(bar, move(name_)) {
+    string socket_path = env_util::get("DWM_SOCKET");
+    if (socket_path.empty()) {
+      // Defined by cmake
+      socket_path = DWM_SOCKET_PATH;
+    }
+
+    if (!file_util::exists(socket_path)) {
+      throw module_error("Could not find socket: " + (socket_path.empty() ? "<empty>" : socket_path));
+    }
+
+    try {
+      m_ipc = factory_util::unique<dwmipc::Connection>(socket_path);
+    } catch (const dwmipc::IPCError& err) {
+      throw module_error(err.what());
+    }
+    m_log.info("%s: Connected to dwm socket", name());
+
+    // Load configuration
+    m_formatter->add(
+        DEFAULT_FORMAT, DEFAULT_FORMAT_TAGS, {TAG_LABEL_TAGS, TAG_LABEL_LAYOUT, TAG_LABEL_FLOATING, TAG_LABEL_TITLE});
+
+    // Populate m_state_labels map with labels and their states
+    if (m_formatter->has(TAG_LABEL_TAGS)) {
+      m_state_labels.insert(
+          std::make_pair(state_t::FOCUSED, load_optional_label(m_conf, name(), "label-focused", DEFAULT_STATE_LABEL)));
+      m_state_labels.insert(std::make_pair(
+          state_t::UNFOCUSED, load_optional_label(m_conf, name(), "label-unfocused", DEFAULT_STATE_LABEL)));
+      m_state_labels.insert(
+          std::make_pair(state_t::VISIBLE, load_optional_label(m_conf, name(), "label-visible", DEFAULT_STATE_LABEL)));
+      m_state_labels.insert(
+          std::make_pair(state_t::URGENT, load_optional_label(m_conf, name(), "label-urgent", DEFAULT_STATE_LABEL)));
+      m_state_labels.insert(
+          std::make_pair(state_t::EMPTY, load_optional_label(m_conf, name(), "label-empty", DEFAULT_STATE_LABEL)));
+    }
+
+    m_seperator_label = load_optional_label(m_conf, name(), "label-separator", "");
+
+    if (m_formatter->has(TAG_LABEL_LAYOUT)) {
+      m_layout_label = load_optional_label(m_conf, name(), "label-layout", "%symbol%");
+    }
+
+    if (m_formatter->has(TAG_LABEL_FLOATING)) {
+      m_floating_label = load_optional_label(m_conf, name(), "label-floating", "");
+    }
+
+    if (m_formatter->has(TAG_LABEL_TITLE)) {
+      m_title_label = load_optional_label(m_conf, name(), "label-title", "%title%");
```

Comment:
How is this different from the `xwindow` module?

Reply:
This title label holds the title of the currently focused window on the **bar's monitor**, whereas the `xwindow` module only shows the currently focused window title on the currently focused monitor. This title label allows unfocused monitors to display their focused window's title.

Reply:
Neat :)